### DISCUSSION TITLE
Dimmer support in a KNX context

### DISCUSSION THREAD

**Discussion by gr6803** [arendst/Tasmota#11291] [Q&A] [answered]
Hi,
I'm playing around with Tasmota's dimming functionality in a KNX environment. Using a tasmota flashed Shelly Dimmer 2 I got dimming commands from KNX working, using Tasmota rules:

_rule1 on event#knxrx_val1 do var1 %value% endon on event#knxrx_val1 do dimmer %value% endon_

Now I'm looking for a method to reply the current brightness level to KNX. Is there a variable as "dimmer#brightness" to use like this:

_on event#knxrx_val2 do knxtx_val2 dimmer#brightness endon_

I did'nt find anything working. Any hints would be appreciated.

Much more helpfull would be a more sophisticated method: having commands available in the Configure KNX Menu to connect to KNX Group Addresses for relative and absolute dimming as standard KNX dimming actuators do:

- brightness level (0...100%, KNX-->Tasmota, absolute dimming) 
- reply brightness level (0...100%, Tasmota -->KNX, absolute dimming)
- step dimming (4bit format, KNX-->Tasmota, relative dimming)

Thanks for your attention.

**Comment by sfromis** [comment, marked as answer]
No experience with KNX, but any rule should be similar to the first format you use:
`on event#knxrx_val1 do dimmer %value% endon`
This means that only `%value%` works after `do`, you should have a trigger (possibly based on JSON) pointing at the field you want.

**Reply by gr6803** [reply]
ok, when for example a dimmer command is issued, then on consol level I see a message similar like this:

_stat/tasmota_D61DE1/RESULT = {"POWER":"ON","Dimmer":30}_ 

The value 30 is what I'm looking for. How do I put this into the variable %value%?

**Reply by ascillato** [reply]
Hi, this is already requested on https://github.com/arendst/Tasmota/issues/7438#issuecomment-609913287

I'm working on it. But you can use rules as a workaround until this feature is added.

**Reply by sfromis** [reply]
Now, you're asking for something entirely different, not specific to KNX. As there is a built-in trigger for dimmer, you should be able to capture that using a rule like:
`on Dimmer#state do var1 %value% endon`

**Reply by gr6803** [reply]
Thanks guys, I got it working. I was not aware of the fact, that "on Dimmer#state" writes the brightness level to %value%. My rule is now as follows for relative dimming (brighter):

_Rule2 on event#knxrx_val2 do dimmer + endon on Dimmer#state do knxtx_val1 %value% endon_

The brightness is replied via TX1.
